src/main/java/com/codefactorygroup/betting/exception/ControllerAdvisor.java

```diff
@@ -0,0 +1,28 @@
+package com.codefactorygroup.betting.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class ControllerAdvisor extends ResponseEntityExceptionHandler {
+
+    @ResponseStatus(
+            value = HttpStatus.NOT_FOUND,
+            reason = "No such participant exists.")
+    @ExceptionHandler(NoSuchParticipantExistsException.class)
+    public void handleNoSuchParticipantExistsException(NoSuchParticipantExistsException ex,
```

Comment:
Rethink please error handling mechanism using a custom ErrorDto , with fields of reaseon , statusCode ...